src/viking.js

```diff
@@ -1,11 +1,92 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health
+        this.strength = strength
+    }
 
-// Viking
-class Viking {}
+    attack() {
+     return this.strength
+    }
+    receiveDamage(damage) {
+     this.health -= damage   
+    }
+}
 
-// Saxon
-class Saxon {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name
+    }
+    receiveDamage(damage) {
+        this.health -= damage
+        let dead = this.health < 0
+        if (Viking !== dead) {
+            console.log(`${Viking.name} has received ${damage} points of damage`)
+        } else {
+            console.log(`${Viking.name} has died in act of combat`)
+        }
+    }
+    battleCry() {
+        return "Odin Owns You All!"
+    }
+}
+
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        Saxon.health -= damage
+        let dead = this.health < 0
+        if (Saxon === dead) {
+            console.log('A Saxon has died in combat')
+        } else {
+            console.log(`A Saxon has received ${damage} points of damage`)
+        }
+
+    attack() {
+        return Saxon.strength
+    }
+}
+}
 
 // War
-class War {}
+class War {
+    constructor (vikingArmy, saxonArmy)
+    this.vikingArmy = []
+    this.saxonArmy = []
+
+    function addViking(this.Viking) {
+        return this.Viking + 1
+    }
```

Comment:
Estamos usando métodos dessa classe War, então não usaremos 'function', faremos da mesma maneira que os métodos que vc fez até aqui: 
addViking(viking), fazendo um push ao vikingArmy, colocando cada vez mais um viking: this.vikingArmy.push(viking) 
vc dessa maneira direciona que está incluindo um viking ao array vikingArmy. 
A mesma coisa para o método addSaxon, que recebe um saxon.